startup/27-vstream-cam.py

```diff
@@ -69,15 +71,30 @@ def trigger(self, *args, **kwargs):
             resource_path=str(Path(self._assets_dir) / Path(data_file)),
             resource_kwargs={},
         )
-        # now discard the start uid, a real one will be added later
-        self._resource_document.pop("run_start")
-        self._asset_docs_cache.append(("resource", self._resource_document))
 
-        data_file = str(
+        self._data_file = str(
             Path(self._resource_document["root"])
             / Path(self._resource_document["resource_path"])
         )
 
+        # now discard the start uid, a real one will be added later
+        self._resource_document.pop("run_start")
+        self._asset_docs_cache.append(("resource", self._resource_document))
+
+        logger.debug(f"{self._data_file = }")
+
+        self._h5file_desc = h5py.File(self._data_file, "w")
+        group = self._h5file_desc.create_group("/entry")
+        self._dataset = group.create_dataset("averaged",
+                                             data=np.zeros((1, *self._frame_shape)),
+                                             maxshape=(None, *self._frame_shape),
```

Comment:
I would make sure that the chunk size is picked to be reasonable rather than letting h5py guess  (aim for like ~1M of data I think).  I would also set a fill value of `np.nan`.

Reply:
no, the only way to specify it is in shape like this,   You might get better performance with more frames per chuck, but that would need to be bench marked.

Fewer chunks will reduce the amount of overhead and may let the compression work better (by giving it a bigger chuckn of data to work on).